system/platform-core/src/test/java/org/platformlambda/core/GenericTypeTest.java

```diff
@@ -114,8 +113,12 @@ public void rejectMixedTypes() {
         list.add(2);
         list.add(pojo2);
         EventEnvelope event = new EventEnvelope();
-        IllegalArgumentException ex = Assert.assertThrows(IllegalArgumentException.class, () -> event.setBody(list));
-        Assert.assertEquals("Unable to serialize because it is a list of mixed types", ex.getMessage());
+        event.setBody(list);
+        List<Object> pojoList = event.getBodyAsListOfPoJo(PoJo.class);
+        Assert.assertEquals(pojoList.getFirst().getClass(), PoJo.class);
```

Comment:
Use static import to make the code less repetitive.
Add assert message that makes tests explanatory.